fdbserver/ptxn/TLogPeekCursor.actor.h

```diff
@@ -479,6 +478,7 @@ class BroadcastedStorageTeamPeekCursorBase : public ptxn::details::VersionSubseq
 	                                     const Version& version_ = 0)
 	  : pCursorContainer(std::move(pCursorContainer_)),
 	    remoteMoreAvailableSnapshot{ false, version_, std::move(pCursorContainerSnapshot_) }, currentVersion(version_) {
```

Comment:
We have 3 levels of cursors: MutableTeamPeekCursor -> BroadcastedStorageTeamPeekCursor -> StorageTeamPeekCursor, all take `beginVersion` as an input. However, `MutableTeamPeekCursor` and `StorageTeamPeekCursor` 's begin version is the version that the cursor doesn't know and need to find out, whereas `BroadcastedStorageTeamPeekCursor`'s begin version is it's current known version.

Can we make them consistent (by changing BroadcastedStorageTeamPeekCursor also takes the next unknown version)? BroadcastedStorageTeamPeekCursor::currentVersion then can be initialized using that version -1.

What do you think?

Reply:
This makes sense. I have updated the code.